File: 2019/01/python/main.py
#!/usr/bin/env python

from fuel import Fuel
from sys import argv, exit

def run(test_case):
  f = open('../input.txt', 'r')
  sum = 0
  
  for line in f:
    mass = int(line)
    if test_case == 'part1':
      sum += Fuel.get_requirement(mass)
    else:
      sum += Fuel.get_recursive_requirement(mass)
  
  return sum

if __name__ == '__main__':
  if len(argv) != 2:
    exit('Wrong number of arguments: required 1')
  elif argv[1].lower() not in ['part1', 'part2']:
    exit('Wrong argument. Accepted: [part1, part2]')

  print(run(argv[1].lower()))

File: 2019/01/ruby/main.rb
#!/usr/bin/env ruby

require_relative 'fuel'

arguments = ARGF.argv
if arguments.length != 1
  p 'Wrong number of arguments. Only 1 permitted'
  return
elsif !['part1', 'part2'].include?(arguments.first.downcase)
  p 'Wrong argument. Permitted arguments: part1, part2'
  return
end

input = File.readlines('../input.txt')
amount = input.sum do |mass|
  case arguments.first.downcase
  when 'part1'
    Fuel.get_requirement(mass.to_i)
  when 'part2'
    Fuel.get_recursive_requirement(mass.to_i)
  end
end
p amount
